### DISCUSSION TITLE
How to see Links in Zipkin

### DISCUSSION THREAD

**Discussion by frgauthier** [open-telemetry/opentelemetry-java#4484] [Q&A] [answered]
I've got a basic question about links and how to see and query for them.  

My SpringBoot app uses both automatic and manual instrumentation.  I have a REST endpoint that I post data to.  That data is buffered for a while.  I save off the **SpanContext** with the buffered data.  Eventually, another REST endpoint will pick up all buffered data and I use the original **SpanContext** to create a link.

```java
private void processItem(Item item, SpanContext originalContext) {
  Span span = tracer.spanBuilder("process-item")
                                .addLink(originalContext)
                                .setSpanKind(SpanKink.INTERNAL)
                                .startSpan();

  try {
    span.setAttribute("item.number", item.getItemNumber());

    // do some processing on the item...
  }
  finally {
    span.end();
  }
}
```

Here's my environment variables:
```bash
export OTEL_SERVICE_NAME=Item Service
export OTEL_TRACES_EXPORTER=zipkin
export OTEL_EXPORTER_ZIPKIN_ENDPOINT=http://localhost:9411/api/v2/spans
export OTEL_PROPAGATORS=tracecontext,baggage
```

In Zipkin, I can see the traces/spans like I mostly expect.  The attributes are visible in Zipkin, however I would have expected to see the link also in Zipkin.  Have I done something wrong, or is Zipkin incapable of showing Links?  If Zipkin is the wrong service to be able to see links, what's a better alternative?

Thanks!

**Comment by mateuszrzeszutek** [comment, marked as answer]
Hey @frgauthier ,
Unfortunately Zipkin does not support span links. You can try using the OTLP exporter, or the Jaeger exporter instead -- both of them are capable of exporting span links.

**Reply by frgauthier** [reply]
Thanks a lot, I can confirm that the Jaeger exporter does export span links.  

Just in case anyone needs a script to start a SpringBoot app with a Jaeger exporter, here you go:

```bash
#! /bin/bash

export OTEL_SERVICE_NAME=My Service
export OTEL_TRACES_EXPORTER=jaeger
export OTEL_EXPORTER_JAEGER_ENDPOINT=http://localhost:14250
export OTEL_EXPORTER_JAEGER_TIMEOUT=10000
export OTEL_PROPAGATORS=tracecontext,baggage

# Replace 1.12.1 with the version of the OTEL javaagent you have
java -javaagent:/path/to/opentelemetry-javaagent-1.12.1.jar -jar MyServiceApp.jar
```

Also, if you want to start a Docker container running Jaeger, here's the magic:

```bash
docker run -d --name jaeger \
  -e COLLECTOR_ZIPKIN_HOST_PORT=:9411 \
  -p 5775:5775/udp \
  -p 6831:6831/udp \
  -p 6832:6832/udp \
  -p 5778:5778 \
  -p 16686:16686 \
  -p 14250:14250 \
  -p 14268:14268 \
  -p 14269:14269 \
  -p 9411:9411 \
  jaegertracing/all-in-one:latest
```
